### DISCUSSION TITLE
No alert from testmynids.org

### DISCUSSION THREAD

**Discussion by Sazerak** [Security-Onion-Solutions/securityonion#2536] [Unsupported Versions] [answered]
Tried running curl testmynids.org/uid/index.html from multiple hosts (including SO itself), also tried testmyids.com and testmyids.ca. Not seeing anything under alerts, hunt, or kibana. I checked if the rule is enabled as per the docs, but the docs don't really say how to tell if the rule is enabled. The result I get is:

alert ip any any -> any any (msg:"GPL ATTACK_RESPONSE id check returned root";
content:"uid=0|28|root|29|"; classtype:bad-unknown; sid:2100498; rev:7; metadata:
created_at 2010_09_23, updated_at 2010_09_23;)

I'm assuming this means that the rule is enabled, but without a reference, I can't be sure. If the rule is enabled, what else could be preventing the alert from generating? Is there any other way I can confirm that SO is working properly? I ran so-test during my previous installation and it ran fine, but it can't be run on 2.3.21 right now.

Edit: I'd like to add that SO may not actually be monitoring my network (at least on the alerts tab). I get lots of results in Kibana from lots of my internal IPs, but if I perform an nmap scan for example, the only time an alert will pop up is if I scan the host running SO. If I scan any other host on my network, I don't get an alert. Any help diagnosing this would be appreciated.

**Comment by weslambert** [comment]
You can check to see if a NIDS rule is enabled by looking in `/opt/so/conf/rules/nids/all.rules` to see if it is remarked or not.

You may want to double-check to see if you are actually monitoring the traffic you intend to be monitoring -- you can check the monitor interface with something like:

`sudo tcpdump -nni bond0`

**Reply by Sazerak** [reply]
Thanks for the reply. I ran the tcpdump command, and it seems to be showing traffic from much of my network (similarly to what I'd see in Wireshark). As for the rules, what do you mean by them being "remarked". If I pick a rule at random, here's what it looks like:

# alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"ET MOBILE_MALWARE SymbOS/Yxes.I PropertyFile.jsp CnC Server Communication"; flow:established,to_server; content:"/PropertyFile.jsp?Version="; nocase; http_uri; content:"&PhoneType="; nocase; http_uri; content:"&PhoneImei="; nocase; http_uri; content:"&PhoneImsi="; nocase; http_uri; reference:url,www.fortiguard.com/encyclopedia/virus/symbos_yxes.i!worm.html; classtype:command-and-control; sid:2012851; rev:3; metadata:created_at 2011_05_25, former_category MOBILE_MALWARE, updated_at 2011_05_25;)

What am I looking for to know if the rule is actually working or not?

**Reply by weslambert** [reply]
If it is not commented out (`#`), it is enabled.  The only way you can know if each rule is "working" as intended is by replaying or monitoring traffic relative to what the rule is looking for.

**Reply by Sazerak** [reply]
hmm.. ok, so that brings me back to my original problem. The rule for testing the IDS (sid:2100498) isn't commented out, but the alert doesn't show up. What would the next step be in order to troubleshoot this? And actually, even though tcpdump is showing plenty of results, the alerts page on SO doesn't seem to show anything that happens outside of the SO host itself. That being said, running a curl command to testmynids from SO doesn't even trigger an alert.

**Comment by defensivedepth** [comment, marked as answer]
Is this a network or ISO install? What kind of deployment (Eval, Standalone, etc) ?

Do you have a TAP or SPAN connected and pointed at the SO monitor NIC?

**Reply by Sazerak** [reply]
Hi Josh! This is an ISO installed as Standalone. It's installed as a VM on my server with ESXi 6.7 as the OS, so what I did was use a separate NIC for the monitoring interface. From there, I created a vSwitch on that NIC and set it to Promiscuous mode (not sure if that's really necessary, but I've read that some people will do this).

I'm not very familiar with the concept of a TAP interface, although I know that when I run VMs on my desktop through Workstation, the TAPs are used as sort of virtual network adapters, so I'm guessing that since I'm using a separate NIC and vSwitch on ESXi, I probably don't need a TAP. Is that right?

**Reply by Sazerak** [reply]
So looks like it was actually related to port mirroring. I enabled port mirroring on my switch and now I can see the GPL ATTACK_RESPONSE from testmynids. I'll have to do some more research on port forwarding though as right now I can only see traffic sent to the downstream switch. I'll have to find out how to cascade the mirror from the upstream switch in order to see traffic from anywhere in my network.

**Reply by defensivedepth** [reply]
Great to hear you got it figured out!

**Reply by Sazerak** [reply]
Well I wouldn't say figured out, lol but I'm well on my way. At least now the topic of this discussion is resolved, but I'm still not 100% convinced it's seeing everything it should. One thing I noticed when looking at the PCAPS is that it only seems to be logging inbound traffic, nothing outbound. Not sure if that's due to a setting in SO or something with my port mirroring again. Back to the drawing board!
